dialer.go

```diff
@@ -97,6 +98,11 @@ type Dialer struct {
 	iamTokenSource oauth2.TokenSource
 }
 
+var (
+	errUseTokenSource    = errors.New("use WithTokenSource when IAM AuthN is not enabled")
+	errUseIAMTokenSource = errors.New("use WithIAMAuthNTokenSources instead of WithTokenSource be used when IAM AuthN is enabled")
```

Comment:
just curious as to why `WithTokenSource` is singular while ``WithIAMAuthNTokenSources` is plural? Is there distinct reasoning for this?

Reply:
WithTokenSource expects a token with SQL Admin scope. WithIAMAuthNTokenSources expects two tokens: one with SQL Admin scope for Admin API operations and one with SQL Service Login scope for logging into the database as an IAM user. We require two separate token sources to avoid embedding an OAuth2 token with admins scope into the ephemeral certificate just in case in the very unlikely case the client certificate is leak (again extremely unlikely given this is all in memory).

Reply:
Thanks! That makes sense